### ISSUE TITLE
Trailing commas required false positive in type declarations

### ISSUE THREAD

**Issue by sushain97** [palantir/tslint#2949]
### Bug Report

- __TSLint version__: 5.4.3
- __TypeScript version__: ~2.2.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
declare module "re-reselect" {
    export type Selector<S, R> = (state: S, ...args: any[]) => R;
    export type Resolver<S> = (state: S, ...args: any[]) => number | string;
    export type OutputSelector<S, T> = (state: S, ...args: any[]) => T;
    export type CachedSelector<S, T> = (resolver: Resolver<S>) => OutputSelector<S, T>;

    // Two selectors
    export default function createCachedSelector<S, R1, R2, T>(
        selectors: [
            Selector<S, R1>,
            Selector<S, R2>
        ],
        combiner: (res1: R1, res2: R2) => T,
    ): CachedSelector<S, T>;
}
```

with `tslint.json` configuration:

```json
{
  "extends": [
    "tslint:latest",
    "tslint-react"
  ],
  "rules": {
    "array-type": [true, "array"],
    "arrow-parens": [true, "ban-single-arg-parens"],
    "ban": [true, ["Object", "assign", "Object.assign does not work in IE11. Use the spread operator instead (or _.assign if you really need it)."]],
    "interface-name": [true, "never-prefix"],
    "jsx-alignment": true,
    "jsx-no-lambda": true,
    "jsx-no-multiline-js": false,
    "jsx-no-string-ref": true,
    "max-classes-per-file": [true, 5],
    "max-line-length": [true, 140],
    "no-debugger": true,
    "no-string-throw": true,
    "object-literal-key-quotes": [true, "as-needed"],
    "object-literal-sort-keys": false,
    "only-arrow-functions": [false],
    "ordered-imports": [false],
    "prefer-const": true,
    "semicolon": [true, "always", "ignore-interfaces", "ignore-bound-class-methods"],
    "trailing-comma": [true, {"multiline": "always", "singleline": "never"}],
    "variable-name": ["check-format", "allow-leading-underscore", "allow-pascal-case"],
    "no-empty-interface": false, /* throws false errors */
    "jsx-wrap-multiline": false
  }
}
```

#### Actual behavior

> src/shared/types/re-reselect.d.ts[11, 27]: Missing trailing comma

#### Expected behavior

No error since Typescript does not allow trailing commas in type declarations: https://github.com/Microsoft/TypeScript/issues/16152

**Comment by JoshuaKGoldberg** [comment]
TypeScript does allow these now - and I can't reproduce this error in TSLint 5.11.0. Resolving as aged away! 🎉
